packages/azure/src/azure-openai-provider.ts

```diff
@@ -80,9 +98,23 @@ export function createAzure(
       fetch: options.fetch,
     });
 
+  const createEmbeddingModel = (
+    modelId: string,
+    settings: OpenAIEmbeddingSettings = {},
+  ) =>
+    new OpenAIEmbeddingModel(modeId, settings, {
+      provider: 'azure-openai.embeddings',
+      headers: getHeaders,
+      url: ({ path }) =>
+        `https://${getResourceName()}.openai.azure.com/openai/deployments/${modelId}${path}?api-version=2024-05-01-preview`,
+      compatibility: 'compatible',
+      fetch: options.fetch,
+    });
+
   const provider = function (
     deploymentId: string,
-    settings?: OpenAIChatSettings,
+    type: 'chat' | 'completion' | 'embeddings' = 'chat',
+    settings?: OpenAIChatSettings | OpenAIEmbeddingSettings,
```

Comment:
the main entry point should just be for the main language model, i.e., the above 2 lines can be reverted

Reply:
Resolved in: 4fd114787b4ed8527734d46ac5e9a2786cebbc49